examples/goldilocks_models/initial_guess.cc

```diff
@@ -0,0 +1,195 @@
+    MatrixXd gamma_scale = get_gamma_scale(gamma_dimension, min_sl, max_sl, min_gi, max_gi, min_tr, max_tr);
+//    initialize variables used for setting initial guess
+    VectorXd initial_guess;
+    VectorXd weight_theta;
+    MatrixXd w_theta;
+    int past_iter;
+    int sample_num;
+//    get theta of current iteration and task of current sample
+    VectorXd current_theta = readCSV(directory + to_string(iter) + string("_theta_s.csv"));
+    MatrixXd current_ground_incline = readCSV(directory + to_string(iter) + string("_") + to_string(sample)
+                                              + string("_ground_incline.csv"));
+    MatrixXd current_stride_length = readCSV(directory + to_string(iter) + string("_") + to_string(sample)
+                                             + string("_stride_length.csv"));
+    MatrixXd current_turning_rate = readCSV(directory + to_string(iter) + string("_") + to_string(sample)
+                                             + string("_turning_rate.csv"));
+    VectorXd current_gamma(gamma_dimension);
+    current_gamma << current_ground_incline(0, 0), current_stride_length(0, 0), current_turning_rate(0,0);
+//    For iter = 1, theta_1 is the same with theta_0. So use the results of iter_0.
+    if(iter == 1)
+    {
+        double distance_gamma_min = 100000; //initialize with a large one
+        int sample_near = 0;
+        for (sample_num = 0; sample_num < total_sample_num; sample_num++) {
+//                check if this sample is success
+            int is_success = (readCSV(directory + to_string(iter-1) + string("_")
+                                      + to_string(sample_num) + string("_is_success.csv")))(0,0);
+            if(is_success == 1) {
+                MatrixXd past_ground_incline = readCSV(directory + to_string(iter-1) + string("_")
+                        +to_string(sample_num) + string("_ground_incline.csv"));
+                MatrixXd past_stride_length = readCSV(directory + to_string(iter-1) + string("_")
+                        + to_string(sample_num) + string("_stride_length.csv"));
+                MatrixXd past_turning_rate = readCSV(directory + to_string(iter-1) + string("_")
+                        + to_string(sample_num) + string("_turning_rate.csv"));
+                VectorXd past_gamma(gamma_dimension);
+                past_gamma << past_ground_incline(0, 0), past_stride_length(0, 0), past_turning_rate(0,0);
+
+                // normalize gamma
+                VectorXd normalize_scale = VectorXd::Zero(gamma_dimension);
+                if(! (min_gi==max_gi))
+                {
+                    normalize_scale[0] = 1/(max_gi-min_gi);
+                }
+                if(! (min_sl==max_sl))
+                {
+                    normalize_scale[1] = 1/(max_sl-min_sl);
+                }
+                if(! (min_tr==max_tr))
+                {
+                    normalize_scale[2] = 1/(max_tr-min_tr);
+                }
```

Comment:
I think you can probably lump this into `gamma_scale`, so that you don't have two scaling factors (`normalize_scale` and `gamma_scale`) in the equation below

Reply:
I separate the normalize scale here because in the last week, I hoped to use different polynomials of the norm to optimize the weights. I can move it back into the scale function.